internal/datastore/postgres/xid8.go

```diff
@@ -0,0 +1,204 @@
+// Adapted from https://github.com/jackc/pgtype/blob/f59f1408937ef0bed249f2bfbafb77222bb48f65/xid.go
+
+package postgres
+
+import (
+	"database/sql/driver"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/jackc/pgio"
+	"github.com/jackc/pgtype"
+)
+
+type pguint64 struct {
+	Uint   uint64
+	Status pgtype.Status
+}
+
+var errUndefined = errors.New("cannot encode status undefined")
+
+type XID8 pguint64
```

Comment:
Mind adding a doc comment explaining this here (or make it private to the package)?